resources/measures/advanced_rtu_control/measure.rb

```diff
@@ -0,0 +1,335 @@
+      puts('Sizing run for Hardsize model failed, cannot hard-size model.')
+      puts("directory: #{Dir.pwd}")
+      return false
+    end
+
+    #apply sizing values
+    model.applySizingValues
+	
+
+   	model.getAirLoopHVACs.sort.each do |air_loop_hvac|
+	    if air_loop_hvac_unitary_system?(air_loop_hvac) #applying to unitary systems 
+		   #set control type
+			air_loop_hvac.supplyComponents.each do |component|#more efficient way of doing this? 
+				obj_type = component.iddObjectType.valueName.to_s
+			    case obj_type
+                when 'OS_AirLoopHVAC_UnitarySystem'
+					component = component.to_AirLoopHVACUnitarySystem.get
+					component.setControlType('SingleZoneVAV') 
+					#Set overall flow rates for air loop 
+					if air_loop_hvac.autosizedDesignSupplyAirFlowRate.is_initialized
+						des_supply_airflow = air_loop_hvac.autosizedDesignSupplyAirFlowRate.get
+						#puts ("des supply airflow" + "#{air_loop_hvac.name.to_s}" "#{des_supply_airflow}" + "new max" + "#{max_flow*des_supply_airflow}")
+						component.setSupplyAirFlowRateDuringCoolingOperation(max_flow*des_supply_airflow) #Set max based on limit after retrofit 
+						component.setSupplyAirFlowRateDuringHeatingOperation(max_flow*des_supply_airflow) #Set max based on limit after retrofit 
+						component.setSupplyAirFlowRateWhenNoCoolingorHeatingisRequired(min_flow*des_supply_airflow) #Set min based on limit after retrofit 
+					elsif air_loop_hvac.designSupplyAirFlowRate.is_initialized
+						des_supply_airflow = air_loop_hvac.designSupplyAirFlowRate.get
+						#puts ("des supply airflow" + "#{air_loop_hvac.name.to_s}" "#{des_supply_airflow}" + "new max" + "#{max_flow*des_supply_airflow}")
+						component.setSupplyAirFlowRateDuringCoolingOperation(max_flow*des_supply_airflow)
+						component.setSupplyAirFlowRateDuringHeatingOperation(max_flow*des_supply_airflow)
+						component.setSupplyAirFlowRateWhenNoCoolingorHeatingisRequired(min_flow*des_supply_airflow)
+					end 
+				sup_fan = component.supplyFan
+				if sup_fan.is_initialized
+					sup_fan = sup_fan.get
+					#handle fan on off objects; replace FanOnOff with FanVariableVolume 
+                    if sup_fan.to_FanOnOff.is_initialized
+					   puts "fan on off" 
+					   sup_fan = sup_fan.to_FanOnOff.get
+					   pressure_rise = sup_fan.pressureRise()
+					   #Create VS supply fan 
+					   fan = OpenStudio::Model::FanVariableVolume.new(model)
+					   fan.setName("#{air_loop_hvac.name} Fan")
+					   fan.setFanEfficiency(fan_tot_eff) 
+					   fan.setFanPowerMinimumFlowRateInputMethod("Fraction")
+					   fan.setPressureRise(pressure_rise)#keep it the same as the existing fan, since the balance of systems is the same 
+					   fan.setFanPowerMinimumFlowFraction(min_flow_fraction)
+					   fan.setMotorEfficiency(fan_mot_eff) unless fan_mot_eff.nil?
+					   #Add it to the unitary sys
+					   #component.resetSupplyFan()
+					   component.setSupplyFan(fan) 
```

Comment:
What performance curve is being used for the new variable speed fan? OS default? Is this reasonable?

Reply:
Thanks for mentioning that. The E+ default fan curve looked reasonable, but now that I look I see there's a Single Zone VAV fan curve in OS Standards. I'll use that for consistency. I'll follow up with you about the min power %.